### ISSUE TITLE
Need Rumble triggers !!

### ISSUE THREAD

**Issue by slishnevsky** [PipedreamHQ/pipedream#12442]
**Rumble**

![image](https://github.com/PipedreamHQ/pipedream/assets/2932739/1cee8f3e-6727-4b04-bc9a-28e8e995b513)

```
Does Rumble API exist?

Yes, Rumble provides several APIs that developers can use for integrating various functionalities of the platform into their applications. The available APIs include:

Rumble Platform API: This API allows developers to integrate Rumble's video services. It includes endpoints for retrieving videos by ID, searching for videos using filters, and other media-related operations. Authentication is required using an API key​ ([Rumble Player](https://www.rumbleplayer.com/developers/Rumble-Platform-API.html))​​ ([Rumble Player](https://www.rumbleplayer.com/developers/index.html))​.

Rumble Player API: This API is used to control video playback and embed videos on websites or applications. It provides options for player customization and embedding, as well as controlling the playback experience​ ([Rumble Player](https://www.rumbleplayer.com/developers/index.html))​.

RumbleTalk RESTful API: This API is focused on managing RumbleTalk group chats. It offers functionality to control, configure, manage, and design chat rooms, manage users, and handle chat messages programmatically​ ([RumbleTalk](https://dev.rumbletalk.com/rest/))​.

Rumble Cloud API: This API is part of Rumble's cloud services, allowing management of cloud resources such as virtual machines, storage, networks, and more. It provides a comprehensive set of endpoints for managing cloud infrastructure​ ([Rumble Cloud](https://docs.rumble.cloud/guides/cloud_console/API/api_endpoints.html))​​ ([Rumble Cloud](https://docs.rumble.cloud/guides/cloud_console/API/))​.

Each API requires proper authentication, typically through API keys or tokens, and detailed documentation is available for developers to get started.
```

**Comment by sergio-eliot-rodriguez** [comment]
Hi @slishnevsky Thanks for the request.

We have base integration for Rumble https://pipedream.com/apps/rumble

We obviously know Rumble API exists.

Help us out with something. Can you please specify the triggers you are looking for?

Bellow is an example for an app called Bexio with actions and trigger components.

> 
> App
> 
> bexio
> Triggers
> 
>     New Bank Account Transaction
>     New/Updated Company
>     New Sales-Invoice
>     New Order
>     New/Updated Person
>     New Product
>     New Project
>     New Quotes
> 
> Actions
> 
>     Create Manual Entry
>     Create Company
>     Create Person
>     Update Person
>     Upload File to Inbox


Please specify the triggers you need for Rumble in a similar way.

Thanks,
Sergio

**Comment by slishnevsky** [comment]
Thank you, Sergio.
Yes I know that you know, just confirmed with the AI :) since the naming of their APIs is a little confusing.
Well, I would like to have more or less the same triggers as for YouTube, but these two as minimum:
New Video in Channel, New Comment Posted.
Thank you for the code, but since you have no-code triggers for YouTube, it probably would make sense to have no-code triggers for Rumble as well.
Thank you!

**Comment by sergio-eliot-rodriguez** [comment]
Awesome, I love it :)

I'm triaging your request for the component development team to build out those triggers.  We'll keep you posted here.

Cheers!

**Comment by drizzt09** [comment]
@sergio-eliot-rodriguez any update on this?
A trigger for Rumble I would say would be beneficial is the 'is_live' data under 'livestreams' endpoint:

![Image](https://github.com/user-attachments/assets/3497cbf2-4681-46d3-a92d-d704ad7e2c70)

**Comment by sergio-eliot-rodriguez** [comment]
Hi @drizzt09 sorry for the delay; I ve prioriotized your request for the component team to pick this up.

**Comment by drizzt09** [comment]
> @sergio-eliot-rodriguez Just want to confirm - accessing videos and comments would use the Rumble Platform API, which we're still waiting on access to, correct? From 1PW - "as of now we have access to livestreaming api but not to rumble platform api yet".

I personally am not aware of the platform API.
But yes the livestream API exists. 
https://rumblefaq.groovehq.com/help/how-to-use-rumble-s-live-stream-api

The trigger I am requesting "is_live" may be different than OP but "is_live" is part of the existing Livestream api

Although at the bottom of the link above shows some info about live chat messages (maybe not comments though).

Please note:

The API URL includes your user UI as well as your live stream key. Authentication is not required for this version of the API.  Please only share this URL to trusted third party sources. You can reset your URL to revoke access any time.
All of the values under "livestreams" including chat messages and rants are only populated during a live stream.
A maximum of 50 results total are output for recent chat messages and rants

**Comment by sergio-eliot-rodriguez** [comment]
@michelle0927 , you are correct -- accessing videos and comments would use the Rumble Platform API for which we don't have access. however @drizzt09 is correct too, he is asking about a new trigger based on LiveStream API for which we have access... I Created a new issue for this #16272, can you handle that one there?

and then let's levae this blocked as Rumble provides API access too. I checked different post on Reddit communities saying Rumble hasn't been getting back on access requests to Platform API, so let's keep this on hold for now.

**Comment by sergio-eliot-rodriguez** [comment]
Hi @drizzt09 , we've completed the new live stream trigger, as requested, you can check it out at: https://pipedream.com/apps/rumble#popular-rumble-triggers

**Comment by drizzt09** [comment]
Thank you. I will give it a test this weekend.
